### ISSUE TITLE
Problem referring to multiple slices

### ISSUE THREAD

**Issue by ametke** [FHIR/sushi#207]
I am creating a profile that inherits from the Genomics IG genomics-report profile and is trying to set mustSupport = true for several result slices that already exist in the parent profile. This results in a NPE when running the IG publisher. Here is a minimal example that can be used to replicate the problem:

```
Profile: Interpretation
Parent: http://hl7.org/fhir/uv/genomics-reporting/StructureDefinition/genomics-report
Id: Interpretation
Title: "Interpretation"
Description: "Represents the interpretation of a genomic analysis, such as the report from a diagnostic laboratory."
* result[gen-grouper] MS
* result[overall] MS
```
The problem seems to be that when more than one slice is referenced, the IG publisher code expects the `sliceName` attribute to be set in the differential and the Sushi compiler does not create it. I also tried adding it explicitly using the ^ syntax but Sushi excludes it because it realises it is the same as the attribute in the parent element definition.

```
Profile: Interpretation
Parent: http://hl7.org/fhir/uv/genomics-reporting/StructureDefinition/genomics-report
Id: Interpretation
Title: "Interpretation"
Description: "Represents the interpretation of a genomic analysis, such as the report from a diagnostic laboratory."
* result[gen-grouper] ^sliceName = "gen-grouper"
* result[gen-grouper] MS
* result[overall] ^sliceName = "overall"
* result[overall] MS
```

I think it is reasonable for the IG publisher code to expect the name of the slice because it shouldn't rely only on the element definition id, and without the `sliceName` attribute this would be the only way to identify each slice.

Attached is also a project that can be used to replicate the issue.

[fsh-issue.zip](https://github.com/FHIR/sushi/files/4178996/fsh-issue.zip)

**Comment by cmoesel** [comment]
Thanks for reporting this @ametke.  You're correct -- our logic to determine what goes in the differential is based on what is _different_ from the base profile (or resource).  Given that the slicename already exists in the base, we're not putting it in the differential.  To be honest, we didn't realize this was a problem for the publisher (which is why we did not have special case logic around it).  We'll look into it!  Thanks!

**Comment by cmoesel** [comment]
I've also confirmed on Zulip that the snapshot generator does not use `id` at all -- so we should do as @ametke suggests and always put the `sliceName` in the differential.

**Comment by ngfreiter** [comment]
Fixed by #212
